sdk/eventhub/azure-eventhub/azure/eventhub/_buffered_producer/_buffered_producer.py

```diff
@@ -85,39 +85,40 @@ def put_events(self, events, timeout_time=None):
         # Put single event or EventDataBatch into the queue.
         # This method would raise OperationTimeout if the queue does not have enough space for the input and
         # flush cannot finish in timeout.
-        try:
-            new_events_len = len(events)
-        except TypeError:
-            new_events_len = 1
-        if self._max_buffer_len - self._cur_buffered_len < new_events_len:
-            _LOGGER.info(
-                "The buffer for partition %r is full. Attempting to flush before adding %r events.",
-                self.partition_id,
-                new_events_len,
-            )
-            # flush the buffer
-            with self._lock:
-                self.flush(timeout_time=timeout_time)
-        if timeout_time and time.time() > timeout_time:
-            raise OperationTimeoutError(
-                "Failed to enqueue events into buffer due to timeout."
-            )
-        try:
-            # add single event into current batch
-            self._cur_batch.add(events)
-        except AttributeError:  # if the input events is a EventDataBatch, put the whole into the buffer
-            # if there are events in cur_batch, enqueue cur_batch to the buffer
-            if self._cur_batch:
+        with self._lock:
+            try:
+                new_events_len = len(events)
+            except TypeError:
+                new_events_len = 1
+            if self._max_buffer_len - self._cur_buffered_len < new_events_len:
+                _LOGGER.info(
+                    "The buffer for partition %r is full. Attempting to flush before adding %r events.",
+                    self.partition_id,
+                    new_events_len,
+                )
+                # flush the buffer
+                with self._lock:
```

Comment:
Question - what does using the same lock twice do from a operational standpoint? Wouldn't we already have the lock to flush since we are in the first with statement.

Reply:
good catch, there should only be one lock. No need for lockception :)